docs/USER_RECORD.md

```diff
@@ -0,0 +1,932 @@
+user. When logging in
+[`pam_systemd`](https://www.freedesktop.org/software/systemd/man/pam_systemd.html)
+will automatically initialized the `$LANG` environment variable from this
+string. The string hence should be in a format compatible with this environment
+variable, for example: `de_DE.UTF8`.
+
+`niceLevel` → An integer value in the range -20…19. When logging in
+[`pam_systemd`](https://www.freedesktop.org/software/systemd/man/pam_systemd.html)
+will automatically initialize the login process' nice level to this value with
+this, which is then inherited by all the user's processes, see
+[`setpriority()`](http://man7.org/linux/man-pages/man2/setpriority.2.html) for
+more information.
+
+`resourceLimits` → An object, where each key refers to a Linux resource limit
+(such as `RLIMIT_NOFILE` and similar). Their values should be an object with
+two keys `cur` and `max` for the soft and hard resource limit. When logging in
+[`pam_systemd`](https://www.freedesktop.org/software/systemd/man/pam_systemd.html)
+will automatically initialize the login process' resource limits to these
+values, which is then inherited by all the user's processes, see
+[`setrlimit()`](http://man7.org/linux/man-pages/man2/setrlimit.2.html) for more
+information.
+
+`locked` → A boolean value. If true the user account is locked, the user may
+not log in. If this field is missing it should be assumed to be false,
+i.e. logins are permitted. This field corresponds with the `sp_expire` field of
+`struct spwd` (i.e. the `/etc/shadow` data for a user) being set to zero or
+one.
+
+`notBeforeUSec` → An unsigned 64bit integer value, indicating a time in µs since
+the UNIX epoch (1970) before which the record should be considered invalid for
+the purpose of logging in.
+
+`notAfterUSec` → Similar, but indicates the point in time *after* which logins
+shall not be permitted anymore. This corresponds to the `sp_expire` field of
+`struct spwd`, when it is set to a value larger than one, but provides finer
+granularity.
+
+`storage` → A string, one of `classic`, `luks`, `directory`, `subvolume`,
+`fscrypt`, `cifs`. Indicates the storage mechanism for the user's home
+directory. If `classic` the home directory is a plain directory as in classic
+UNIX. When `directory`, the home directory is a regular directory, but the
+`~.identity` file in it contains the user's user record, so that the directory
+is self-contained. Similar, `subvolume` is a `btrfs` subvolume that also
+contains a `~/.identity` use record; `fscrypt` is an `fscrypt`-encrypted
+directory, also containing the `~/.identity` user record; `luks` is a per-user
+LUKS volume that is mounted as home directory, and `cifs` a home directory
+mounted from a Windows File Share. The five latter types are primarily used by
+`systemd-homed` when managing home directories, but may be used if other
+managers are used too. If this is not set `classic` is the implied default.
```

Comment:
i don't think this should be orthogonal. For "new-style" stuff we should insist on the ~./identity file to make it self-sufficient, and for the "old-style" stuff we should insist on not having it, to be fully compatible, and make sure that programs are never confused into maybe updating it or maybe not... 

or to say this differently: it doesn't make sense to just say "fscrypt" without implying our way to manage "fscrypt" simply because we need to know mechanism and metadata to make "fscrypt" work, because of the key mgmt and so on... Thus "fscrypt" without "~/.identity" is not useful, nobody would then know what to do with that. And that's even more the case for "LUKS" I think. Without the ~/.identity stuff you can't even know whether the password of the user and of the LUKS volume are the same...

Hence, let's draw clear lines here: the classic stuff is the classic stuff the way it always was, and the fancier new stuff always is self descriptive. And if there are people who want fscrypt but no ~/.identity then it's on them to manage it, and then either expose it as "classic", or maybe ping us and ask for a new value to be assigned to their use, which clarifies how the fscrypt is managed. Similar if there are people who want LUKS but no ~/.identity then they need to manage that on their own, and should declare how that's done, and hence should ask for a new value that clarifies all that.